File: tests/fixtures/cpp/user_service.cpp
#include <string>
#include <memory>
#include "repository.hpp"

namespace app {

class User {
public:
    std::string id;
    std::string name;
};

class BaseService {
public:
    virtual ~BaseService() = default;
    void log(const std::string& msg);
};

class UserService : public BaseService {
public:
    UserService(std::shared_ptr<Repository> repo);
    User find(int id);
    User create(const std::string& name);
private:
    std::shared_ptr<Repository> repo_;
};

UserService::UserService(std::shared_ptr<Repository> repo) : repo_(repo) {}

User UserService::find(int id) {
    return repo_->find(id);
}

User UserService::create(const std::string& name) {
    log("creating");
    return User{};
}

int helper(int a, int b) {
    return a + b;
}

}  // namespace app

File: tests/fixtures/php/UserService.php
<?php
namespace App\Service;

use App\Repo\UserRepo;
use App\Entity\User;

interface UserRepository {
    public function find(int $id): ?User;
    public function save(User $user): User;
}

class BaseService {
    protected function log(string $msg): void {}
}

class UserService extends BaseService implements UserRepository {
    private UserRepo $repo;

    public function __construct(UserRepo $repo) {
        $this->repo = $repo;
    }

    public function find(int $id): ?User {
        return $this->repo->find($id);
    }

    public function save(User $user): User {
        $this->log("saving");
        return $this->repo->save($user);
    }
}

function plainHelper(int $a, int $b): int {
    return $a + $b;
}
